towerlib/towerlib.py

```diff
@@ -2018,3 +2034,300 @@ def get_schedule_by_name(self, name):
+        credential_id = None
+        projects = list(self.projects)
+        for scm_credential_type_id in scm_credential_type_ids:
+            for project in projects:
+                if project.scm_url == scm_url:
+                    self._logger.debug("Project found with the scm credential.")
+                    if project.credential.credential_type.id == scm_credential_type_id:
+                        credential_id = project.credential.id
+                        if credential_id:
+                            self._logger.debug("Credential ID found from the existing project '{}' and the id is '{}'"
+                                               .format(project.name, credential_id))
+                            return credential_id
+        return credential_id
+
+    def get_ansible_facts_by_host_id(self, host_id):
+        """Get the ansible_facts of the given host.
+
+        Args:
+            host_id: id of the host for which the method will return the ansible_facts.
+
+        Returns:
+            dict: ansible_facts as dictionary.
+
+        """
+        host = self.get_host_by_id(host_id)
+        if not host:
+            raise InvalidHost(host_id)
+        return host.ansible_facts()
+
+    def get_jobs_by_name(self, name):
+        """Get filtered list of jobs for a given name.
+
+        Args:
+            name: the given job name.
+
+        Returns:
+             list: the filtered list of jobs.
+
+        """
+
+        return self.jobs.filter({'name__iexact': name})
+
+    def get_job_by_id(self, id_):
+        """Retrieves a job by id.
+
+        Args:
+            id_: The id of the job to retrieve.
+
+        Returns:
+            Host: The host if a match is found else None.
+
+        """
+        return next(self.jobs.filter({'id': id_}), None)
+
+
+    def get_project_update_by_id(self, id_):
+        """Retrieves a project_update by id.
+
+        Args:
+            id_: The id of the project_update to retrieve.
+
+        Returns:
+            Host: The project_update if a match is found else None.
+
+        """
+        return next(self.project_updates.filter({'id': id_}), None)
+
+    def get_project_updates_by_name(self, name):
+        """Retrieves project_updates matching a certain name.
+
+        Args:
+            name: the given job_update name.
+
+        Returns:
+             list: the filtered list of project update jobs.
+
+        """
+        return self.project_updates.filter({'name__iexact': name})
+
+    @property
+    def job_events(self):
+        """The job templates configured in tower.
+
+        Returns:
+            EntityManager: The manager object for job templates.
+
+        """
+        return EntityManager(self,
+                             entity_name='job_events',
+                             entity_object='JobEvent',
+                             primary_match_field='name')
+
+    def get_job_creation_dates_by_host(self, host):
```

Comment:
This is very specific client side logic and I can't really see the value in having just a list of dates. If this makes any sense, it should be implemented on the host object as a method and not on this level. Here it is just confusing.